### PR TITLE
Tc1 event link modal

### PR DESCRIPTION
**Pull request by Brianna-Flo**
## Description

<what this pull request is doing>

- create an event class to hold event objects created from the response from the google calendar api
- created a modal to present to the user the names, date, start and end time, and link to each of the created google calendar events

<what will be done in later PRs and not included here>

- further styling of the modal

## Milestones
<the milestones or stories/features that this works towards>

- TC1, integration with google calendar

## Resources
<links to tutorials, code snippets, inspirations>
<if you took or translated specific code from the internet, please call it out here!>

## Test Plan
<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>
<div>
    <a href="https://www.loom.com/share/d21188245e264654932ac8131a85584c">
      <p>Event Info Modal Demo - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/d21188245e264654932ac8131a85584c">
      
    </a>
  </div>

### CODE REVIEW

File: frontend/src/components/calendar/CalendarEventLinkModal.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        <ModalClose
+          onClick={toggleModal}
+          variant="plain"
+          sx={{ zIndex: 2, m: 1 }}
+        />
+        <DialogContent sx={{ my: 3 }}>
+          <TitledListView
+            headerList={[
+              {
+                title: "What would you like to diff?",
```
Comment (nithyad) [inline frontend/src/components/calendar/CalendarEventLinkModal.tsx]: This title is a little confusing imo
